rust/chains/hyperlane-ethereum/src/mailbox.rs

```diff
@@ -225,10 +225,23 @@ where
                 .await?
                 .saturating_add(U256::from(GAS_ESTIMATE_BUFFER))
         };
+        let chain_id = match self.provider.get_chainid().await {
+            Ok(chainId) => chainId.as_u32(),
+            // Couldn't get chainId, assume not 1559
+            Err(_) => return Ok(tx.gas(gas_limit)),
+        };
         let Ok((max_fee, max_priority_fee)) = self.provider.estimate_eip1559_fees(None).await else {
             // Is not EIP 1559 chain
             return Ok(tx.gas(gas_limit))
         };
+        let max_priority_fee =
```

Comment:
flagging my comment here bc it's outdated https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/1952#discussion_r1140154886